### PR TITLE
Bash example CLI

### PR DESCRIPTION
**Pull request by jburel**
Example script showing how to use the CLI commands

cc @pwalczysko

### CODE REVIEW

File: practical/bash/import_as_create_annotation.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Log in as
+$OMEROPATH login --sudo ${SUDOER} -u $OMEUSER -s $HOST -w $PASSWORD
+
+# Create a dataset as the specified user
+dataset=`$OMEROPATH obj new Dataset name='Basel-workflow'`
+
+# Import the image in the newly created dataset
+result=`$OMEROPATH import -T $dataset $IMAGEPATH --output ids`
+
+# Retrieve the Image's id. it is Image:123
```
Comment (mtbc) [inline practical/bash/import_as_create_annotation.sh]: uncapitalized "it" starts a new sentence

File: practical/bash/import_as_create_annotation.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+imageid=`cut -d':' -f2 <<< $result`
+printf 'imageid %s \n' "$id"
+
+# Upload a CSV and attach it to the Image
+result=`$OMEROPATH upload $ATTACHMENTPATH`
+originalfileid=`cut -d':' -f2 <<< $result`
+printf 'originalfileid %s \n' "$originalfileid"
+
+# Create a file annotatin
```
Comment (mtbc) [inline practical/bash/import_as_create_annotation.sh]: annotation

File: practical/bash/import_as_create_annotation.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+fi
+#printf $IMAGEPATH
+
+if [ "$ATTACHMENTPATH"="/" ]; then
+    touch file_to_upload.csv
+    ATTACHMENTPATH=$dir/file_to_upload.csv
+    deletefile=true
+fi
+
+# Log in as
```
Comment (mtbc) [inline practical/bash/import_as_create_annotation.sh]: as ... ?

File: practical/bash/import_as_create_annotation.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+# Create a file annotatin
+result=`$OMEROPATH obj new FileAnnotation file=OriginalFile:$originalfileid`
+fileid=`cut -d':' -f2 <<< $result`
+printf 'fileid %s \n' "$fileid"
+
+# Link the annotation to the Image
+$OMEROPATH obj new ImageAnnotationLink parent=Image:$imageid child=FileAnnotation:$fileid
+
+# delete the Image and the local file if created
```
Comment (mtbc) [inline practical/bash/import_as_create_annotation.sh]: capitalize "Delete" to fit in with above

File: practical/bash/import_as_create_annotation.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+#   with this program; if not, write to the Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+#
+# In this script we:
+#   * log in as another user
+#   * create a new dataset
+#   * import an image
+#   * upload a attachment
```
Comment (pwalczysko) [inline practical/bash/import_as_create_annotation.sh]: ``a attachment`` -> ``an attachment``

File: practical/bash/import_as_create_annotation.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+deleteimage=false
+deletefile=false
+
+if [ "$IMAGEPATH"="/" ]; then
+    touch image_to_import.fake
+    IMAGEPATH=$dir/image_to_import.fake
+    deleteimage=true
+fi
+#printf $IMAGEPATH
```
Comment (pwalczysko) [inline practical/bash/import_as_create_annotation.sh]: do you want to retain this line for further use or is it just a testing remnant ?

File: practical/bash/import_as_create_annotation.sh [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+dataset=`$OMEROPATH obj new Dataset name='Basel-workflow'`
+
+# Import the image in the newly created dataset
+result=`$OMEROPATH import -T $dataset $IMAGEPATH --output ids`
+
+# Retrieve the Image's id. It is returned as Image:123
+imageid=`cut -d':' -f2 <<< $result`
+printf 'imageid %s \n' "$id"
+
+# Upload a CSV and attach it to the Image
```
Comment (pwalczysko) [inline practical/bash/import_as_create_annotation.sh]: I am not sure this paragraph does the ``attach it to the Image`` part.

I guess this is done two paragraphs below, at https://github.com/ome/training-scripts/pull/35/files#diff-54247341f66f1673ff5f0d8697e623abR75

**Comment by pwalczysko** [comment]
The script works fine when the image (and the csv file) are both fake (tested on outreach)


Bug ?
The script does not seem to work on outreach in case I pass real image paths into ``IMAGEPATH``.

For example 

``PASSWORD=xxx IMAGEPATH=/OMERO/in-place-import/svs/77928.svs bash /OMERO/in-place-import/import_as_create_annotation.sh``

ends up with the script uploading a fake image again.

File: practical/bash/import_as_create_annotation.sh [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+$OMEROPATH login --sudo ${SUDOER} -u $OMEUSER -s $HOST -w $PASSWORD
+
+# Create a dataset as the specified user
+dataset=`$OMEROPATH obj new Dataset name='Basel-workflow'`
+
+# Import the image in the newly created dataset
+result=`$OMEROPATH import -T $dataset $IMAGEPATH --output ids`
+
+# Retrieve the Image's id. It is returned as Image:123
+imageid=`cut -d':' -f2 <<< $result`
```
Comment (pwalczysko) [inline practical/bash/import_as_create_annotation.sh]: I guess it is this line which fails with 

``ValueError: invalid literal for long() with base 10: '29320,29321,29322'``

if the image ID returned is not just one number (the svs which I imported consists of three images in OMERO).

I think there could be 2 solutions:

   - parse the multi-ID output, and select just the first ID, and add the attachment to that image
   - parse the ID output only so far as to ascertain whether or not there are multiple IDs returned. If multiple IDs are returned, then inform the user "sorry, multiple images imported, do not know to which attach the file" and stop the script there

**Comment by pwalczysko** [comment]
Works  fine now, the non-fake file import as well as the non-fake attachment upload.

The only problem here is https://github.com/ome/training-scripts/pull/35/files#r213252631 - maybe, to spare time and effort, we could go for the second option suggested in my comment ?

File: practical/bash/import_as_create_annotation.sh [hunk 9]
```diff
@@ -25,14 +25,19 @@
...
+#OMEROPATH=${OMEROPATH:-/opt/omero/server/OMERO.server/bin/omero}
+#PASSWORD=${PASSWORD:-ome}
+#HOST=${HOST:-outreach.openmicroscopy.org}
 SUDOER=${SUDOER:-trainer-1}
-OMEUSER=${OMEUSER:-user-1}
+#OMEUSER=${OMEUSER:-user-1}
 IMAGEPATH=${IMAGEPATH:-}
 ATTACHMENTPATH=${ATTACHMENTPATH:-}
 
+OMEROPATH=${OMEROPATH:-/Users/jmarie/Documents/git-repo/openmicroscopy/dist/bin/omero}
```
Comment (pwalczysko) [inline practical/bash/import_as_create_annotation.sh]: ``/Users/jmarie/...`` cannot be right -> consequently, it does not work on outreach either

File: practical/bash/import_as_create_annotation.sh [hunk 10]
```diff
@@ -25,14 +25,19 @@
...
+PASSWORD=${PASSWORD:-omero}
```
Comment (pwalczysko) [inline practical/bash/import_as_create_annotation.sh]: ?

File: practical/bash/import_as_create_annotation.sh [hunk 11]
```diff
@@ -25,14 +25,19 @@
...
+HOST=${HOST:-localhost}
+OMEUSER=${OMEUSER:-root}
```
Comment (pwalczysko) [inline practical/bash/import_as_create_annotation.sh]: strange

Comment (jburel) [reply practical/bash/import_as_create_annotation.sh]: sorry I did not clean after testing

**Comment by pwalczysko** [comment]
Works nicely now, thank you. I think the tactics of attaching the CSV to all three images is good.

Ready to merge FMPOV

**Pull request merged**